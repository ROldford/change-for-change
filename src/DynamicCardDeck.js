// Makes a single deck given a variable set of cards

import React from 'react';
import { CardDeck } from 'reactstrap';
import CustomCard from './CustomCard';

export default class DynamicCardDeck extends React.Component {
  createCard(data) {
    return <CustomCard data={data} />;
  }

  createBlankCard() {
    const blankCardData = {
      imageSource: 'https://www.countryflags.io/aa/flat/64.png',
      imageAltText: '',
      cardTitle: '',
      cardText: ''
    };
    return <CustomCard data={blankCardData} />;
  }

  createCards(cards) {
    return cards.map(this.createCard);
  }

  render() {
    let cards = this.props.cards;
    while (cards.length < this.props.deckSize) {
      let blankCard = this.createBlankCard();
      cards.push(blankCard);
    }
    return (
      <CardDeck className="my-2">
        {this.createCards(this.props.cards)}
      </CardDeck>
    );
  }
};

DynamicCardDeck.propTypes = {
  cards: PropTypes.shape({
    cardTitle: PropTypes.string,
    cardText: PropTypes.string,
    imageSource: PropTypes.string,
    imageAltText: PropTypes.string
  }),
  deckSize: PropTypes.number.isRequired
}
