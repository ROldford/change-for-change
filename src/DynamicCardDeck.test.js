import DynamicCardDeck from "./DynamicCardDeck";

const setup = () => {
  return [{
    key: "1",
    data: {
      imageSource: 'https://www.countryflags.io/ca/flat/64.png',
      imageAltText: 'Card image cap',
      cardTitle: '1 Card title',
      cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
    }
  },
  {
    key: "2",
    data: {
      imageSource: 'https://www.countryflags.io/ca/flat/64.png',
      imageAltText: 'Card image cap',
      cardTitle: '2 Card title',
      cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
    }
  },
  {
    key: "3",
    data: {
      imageSource: 'https://www.countryflags.io/ca/flat/64.png',
      imageAltText: 'Flag: Canada',
      cardTitle: '3 Card title',
      cardText: 'This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.'
    }
  }];
}

describe('Tests', () => {
  it("renders correctly", () => {
    let cards = setup();
    const wrapper = shallow(
      <DynamicCardDeck cards={cards} deckSize={3} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders blank cards when needed", () => {
    let cards = setup();
    const wrapper = shallow(
      <DynamicCardDeck cards={cards} deckSize={5} />
    )
    expect(wrapper).toMatchSnapshot();
    const foundCards = wrapper.find('CustomCard');
    expect(foundCards.length).toEqual(5);
    // const placeholderCards = foundCards.filterWhere((card) => {
    //   card.props().data.imageSource === 'https://www.countryflags.io/aa/flat/64.png'
    // });
    // expect(placeholderCards.length).toEqual(2);
  });
})
