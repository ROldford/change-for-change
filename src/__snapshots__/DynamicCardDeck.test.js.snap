// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests renders blank cards when needed 1`] = `
<CardDeck
  className="my-2"
  tag="div"
>
  <CustomCard
    data={
      Object {
        "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
        "cardTitle": "1 Card title",
        "imageAltText": "Card image cap",
        "imageSource": "https://www.countryflags.io/ca/flat/64.png",
      }
    }
    key="1"
  />
  <CustomCard
    data={
      Object {
        "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
        "cardTitle": "2 Card title",
        "imageAltText": "Card image cap",
        "imageSource": "https://www.countryflags.io/ca/flat/64.png",
      }
    }
    key="2"
  />
  <CustomCard
    data={
      Object {
        "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
        "cardTitle": "3 Card title",
        "imageAltText": "Flag: Canada",
        "imageSource": "https://www.countryflags.io/ca/flat/64.png",
      }
    }
    key="3"
  />
  <CustomCard
    data={
      Object {
        "cardText": "",
        "cardTitle": "",
        "imageAltText": "",
        "imageSource": "https://www.countryflags.io/aa/flat/64.png",
      }
    }
    key="blank {cardCountdown}"
  />
  <CustomCard
    data={
      Object {
        "cardText": "",
        "cardTitle": "",
        "imageAltText": "",
        "imageSource": "https://www.countryflags.io/aa/flat/64.png",
      }
    }
    key="blank {cardCountdown}"
  />
</CardDeck>
`;

exports[`Tests renders correctly 1`] = `
<CardDeck
  className="my-2"
  tag="div"
>
  <CustomCard
    data={
      Object {
        "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
        "cardTitle": "1 Card title",
        "imageAltText": "Card image cap",
        "imageSource": "https://www.countryflags.io/ca/flat/64.png",
      }
    }
    key="1"
  />
  <CustomCard
    data={
      Object {
        "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
        "cardTitle": "2 Card title",
        "imageAltText": "Card image cap",
        "imageSource": "https://www.countryflags.io/ca/flat/64.png",
      }
    }
    key="2"
  />
  <CustomCard
    data={
      Object {
        "cardText": "This is a longer card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.",
        "cardTitle": "3 Card title",
        "imageAltText": "Flag: Canada",
        "imageSource": "https://www.countryflags.io/ca/flat/64.png",
      }
    }
    key="3"
  />
</CardDeck>
`;
